src/com/lftechnology/java/training/srijan/basics/coreapp/dao/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,305 @@
+				con.close();
+			} catch (SQLException e1) {
+				LOGGER.log(Level.INFO, "SQL Exception{0}", e1);
+			}
+		}
+	}
+
+	public Employee findById(int employeeIdToUpdate) {
+		Connection con = null;
+		ResultSet resultSet = null;
+		con = DbConnection.dbConnection();
+		PreparedStatement prepareStatementUpdateInfo = null;
+		Employee employee = new Employee();
+		try {
+			if (con != null) {
+				String query = "Select * from employeeDetails where id = ?";
+				prepareStatementUpdateInfo = con.prepareStatement(query);
+				prepareStatementUpdateInfo.setInt(1, employeeIdToUpdate);
+				resultSet = prepareStatementUpdateInfo.executeQuery();
+				while (resultSet.next()) {
+					employee.setId(resultSet.getInt("id"));
+					employee.setUserName(resultSet.getString("username"));
+					employee.setFullName(resultSet.getString("fullName"));
+					employee.setDepartment(resultSet.getString("department"));
+					employee.setAddress(resultSet.getString("address"));
+					employee.setIsTerminated(resultSet.getBoolean("is_terminated"));
+				}
+			}
+		} catch (Exception ex) {
+			LOGGER.log(Level.INFO, "Exception:{0}", ex);
+		} finally {
+			try {
+				resultSet.close();
+				prepareStatementUpdateInfo.close();
+				con.close();
+			} catch (SQLException ex) {
+				LOGGER.log(Level.INFO, "SQL Exception {0}", ex);
+			}
+		}
+		return employee;
+	}
+
+	@Override
+	public UserRole loginValidation(Employee e) {
+		ResultSet resultSet = null;
+		Connection con = null;
+		con = DbConnection.dbConnection();
+		PreparedStatement prepareStatementLoginCheck = null;
+		try {
+			if (con != null) {
+				String query = "Select * from userCredential where user_name = ? and password = ?";
+				prepareStatementLoginCheck = con.prepareStatement(query);
+				prepareStatementLoginCheck.setString(1, e.getUserName());
+				prepareStatementLoginCheck.setString(2, e.getPassword());
+				resultSet = prepareStatementLoginCheck.executeQuery();
+				if (resultSet.next()) {
+					LOGGER.log(Level.INFO, "Login Successful");
+					UserRole role = UserRole.valueOf(resultSet.getString("user_role"));
+					return role;
+				} else {
+					LOGGER.log(Level.INFO, "Entry not found in Database");
+				}
+			} else {
+				LOGGER.log(Level.INFO, "Failed to connect to database");
+			}
+		} catch (SQLException e1) {
+			LOGGER.log(Level.INFO, "SQl Exception{0}", e1);
+		} finally {
+			try {
+				resultSet.close();
+				if (prepareStatementLoginCheck != null) {
+					prepareStatementLoginCheck.close();
+				}
+				if (con != null) {
+					con.close();
+				}
+			} catch (SQLException e1) {
+				LOGGER.log(Level.INFO, "SQLException{0}", e1);
+			}
+
+		}
+		return UserRole.INVALID;
+	}
+
+	@Override
+	public List<Employee> viewEmployeeDetails() {
+		Statement statement = null;
+		ResultSet resultSet = null;
+		Connection con = null;
+		List<Employee> employees = new ArrayList<>();
+		con = DbConnection.dbConnection();
+		if (con != null) {
+			System.out.println("inside view employeedetails");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)